quinn/src/send_stream.rs

```diff
@@ -261,6 +261,12 @@ impl tokio::io::AsyncWrite for SendStream {
 impl Drop for SendStream {
     fn drop(&mut self) {
         let mut conn = self.conn.state.lock("SendStream::drop");
+
+        // clean up any previously registered wakers
+        conn.finishing.remove(&self.stream);
+        conn.stopped.remove(&self.stream);
```

Comment:
Provided people aren't leaking things themselves, these are redundant to the `Drop` impls of `Finish` and `Stopped`. I think I actually like them a bit better here than in the futures since it lets us hit the mutex less, but for clarity we shouldn't have them in both places.